### PR TITLE
ocaml: re-establish alphabetical order at top-level

### PR DESCRIPTION
**Pull request by superherointj**
ocaml: re-established alphabetical order

* Considering `_` and `-` equal. And adding it to start (before `a`).

* Added comment banner for letters. Because... uh.. humans get order wrong?

### CODE REVIEW

File: pkgs/top-level/ocaml-packages.nix [hunk 1]
```diff
@@ -996,6 +992,8 @@ let
 
     parmap = callPackage ../development/ocaml-modules/parmap { };
 
+    ocaml_cairo = callPackage ../development/ocaml-modules/ocaml-cairo { };
```
Comment (superherointj) [inline pkgs/top-level/ocaml-packages.nix:995]:
```suggestion
    ocaml_cairo = callPackage ../development/ocaml-modules/ocaml-cairo { };
```
This is still out of order. As it is after `lwt`.

File: pkgs/top-level/ocaml-packages.nix [hunk 2]
```diff
@@ -1054,6 +1052,8 @@ let
 
     ocamlnet = callPackage ../development/ocaml-modules/ocamlnet { };
 
+    ocaml_lwt = lwt;
```
Comment (superherointj) [inline pkgs/top-level/ocaml-packages.nix:1055]:
```suggestion
    ocaml_lwt = lwt;
```
This is still out of order. As it is after `lwt`.

File: pkgs/top-level/ocaml-packages.nix [hunk 3]
```diff
@@ -58,28 +58,34 @@ let
...
+    benchmark = callPackage ../development/ocaml-modules/benchmark { };
+
     bheap = callPackage ../development/ocaml-modules/bheap { };
 
     bigarray-compat = callPackage ../development/ocaml-modules/bigarray-compat { };
 
     bigarray-overlap = callPackage ../development/ocaml-modules/bigarray-overlap { };
 
-    bigstringaf = callPackage ../development/ocaml-modules/bigstringaf { };
+    biniou = callPackage ../development/ocaml-modules/biniou { };
```
Comment (GabrielDougherty) [inline pkgs/top-level/ocaml-packages.nix:77]: `biniou` is out of order, as it is before `bigstring` here and n should come after g.

**Review by GabrielDougherty — CHANGES_REQUESTED** [review]

File: pkgs/top-level/ocaml-packages.nix (hunk 3 above)
Comment (superherointj) [reply pkgs/top-level/ocaml-packages.nix]: Thanks!

**Comment by GabrielDougherty** [comment]
Looks good! It looks like some packages are still out of order before "L":

<details>

<summary>Package list</summary>

- callipyge
- callipyge
- carton
- carton-git
- carton-lwt
- class_group_vdf
- ctypes
- ctypes_stubs_js
- dnssec
- extlib-1-7-9
- ff-pbt
- ff-sig
- frontc
- gettext-camomile
- gettext-stub
- google-drive-ocamlfuse
- hol_light
- http-mirage-client
- janeStreet_0_9_0
- jsonrpc

... list continues

</details>

based on this report I made with grep / cut / sort:
https://www.diffchecker.com/Vwq7yuLQ/

But I assume that's because they were all listed after `lwt` in the current file!

**Review by GabrielDougherty — APPROVED** [review]

**Comment by superherointj** [comment]
I'm working on finishing the whole file. And yes, some of those other names are beyond that part.

**Comment by superherointj** [comment]
@GabrielDougherty Do your magic again.

**Comment by GabrielDougherty** [comment]
Ok, here is the diff generated from revision d279c9f6547b2d148a8390bfa8ff1ed12487756d: https://www.diffchecker.com/qisRGj6Y/

left hand side is the file's current order. Right hand side is the ideal sorted order.

**Comment by superherointj** [comment]
I have updated a few. I think problem is the criteria I used for - and _ and where it goes is what is not matching. But so far, seems good. What do you think?

**Comment by GabrielDougherty** [comment]
Yep, LGTM. I did an additional check to ensure no packages were deleted by this PR and yes, all packages are still there

**Review by GabrielDougherty — APPROVED** [review]

**Comment by superherointj** [comment]
@vbgl Approve?

File: pkgs/top-level/ocaml-packages.nix [hunk 4]
```diff
@@ -102,24 +106,32 @@ let
 
     brr = callPackage ../development/ocaml-modules/brr { };
 
+    buildDunePackage = callPackage ../build-support/ocaml/dune.nix {}; # Libs
+
+    buildOasisPackage = callPackage ../build-support/ocaml/oasis.nix { }; # Libs
+
```
Comment (vbgl) [inline pkgs/top-level/ocaml-packages.nix:112]: Those are better kept out of order (IMHO).

File: pkgs/top-level/ocaml-packages.nix [hunk 5]
```diff
@@ -567,14 +626,26 @@ let
 
     hmap = callPackage ../development/ocaml-modules/hmap { };
 
+    hol_light = callPackage ../applications/science/logic/hol_light { }; # App from all-packages
```
Comment (vbgl) [inline pkgs/top-level/ocaml-packages.nix:629]: I’d rather see this out of this set. Currently, the few packages that should be kicked out of here are conveniently gathered at the end. With this change, they get lost.

**Review by vbgl — COMMENTED** [review]
I don’t like this kind of big change that mostly bring merge conflicts and pollute the history. But you may have good motivations.

File: pkgs/top-level/ocaml-packages.nix (hunk 4 above)
Comment (superherointj) [reply pkgs/top-level/ocaml-packages.nix:112]: Can I add it to end? People won't change it. Less scrolling for adding new packages.

File: pkgs/top-level/ocaml-packages.nix (hunk 5 above)
Comment (superherointj) [reply pkgs/top-level/ocaml-packages.nix]: Done

File: pkgs/top-level/ocaml-packages.nix (hunk 4 above)
Comment (superherointj) [reply pkgs/top-level/ocaml-packages.nix]: Done

**Comment by superherointj** [comment]
I did a rebase and the PR was destroyed.

**Comment by GabrielDougherty** [comment]
You should be able to recover it from git reflog

**Comment by superherointj** [comment]
> You should be able to recover it from git reflog

Great idea!

**Comment by superherointj** [comment]
@GabrielDougherty You have just saved the PR.

**Comment by superherointj** [comment]
I have just manually reviewed it. Seems good. Eval is OK.
I want to merge this ASAP, because on every PR that changes top-level this will likely break.
As soon as ofBorg is happy I'll merge it.

**Comment by superherointj** [comment]
@NixOS/ofborg eval

**Comment by superherointj** [comment]
@ofborg eval

**Pull request merged**